### PR TITLE
Pipeline sink refactor

### PR DESCRIPTION
**Pull request by jonathandturner**
This updates how we sink the pipeline and opens us up to some new features in the future:

* Moves where autoview is assumed out of `run_pipeline`. This means that `run_pipeline` will now output a stream that can be later consumed. In the future, this will allow blocks to hold a pipeline and evaluate it without viewing it.
* Adds a `-c` commandline flag to pass in a pipeline of commands via a string that is run to completions. Note: stdin redirection is not yet supported

### CODE REVIEW

File: src/commands/classified/pipeline.rs [hunk 1]
```diff
@@ -48,26 +46,5 @@ pub(crate) async fn run_pipeline(
         };
     }
 
-    use futures::stream::TryStreamExt;
-    if let Some(input) = input {
-        let mut output_stream: OutputStream = input.into();
-        loop {
-            match output_stream.try_next().await {
```
Comment (thegedge) [inline src/commands/classified/pipeline.rs]: Is the intent for this removal that commands should be handling Ctrl-C?

File: src/commands/autoview.rs [hunk 2]
```diff
@@ -170,3 +201,25 @@ pub fn autoview(
...
+                    span,
+                )),
+                positional: None,
+                named: None,
+                span,
+            },
+            source: context.source.clone(),
+            name_tag: context.name.clone(),
+        },
+    }
```
Comment (thegedge) [inline src/commands/autoview.rs]: 😅 reading this part makes me realize there are a lot of concepts I still need to learn.

Comment (jonathandturner) [reply src/commands/autoview.rs]: It's, umm, not the prettiest thing admittedly :sweat_smile:

File: src/commands/classified/pipeline.rs (hunk 1 above)
Comment (jonathandturner) [reply src/commands/classified/pipeline.rs]: What I'm doing here probably need more explanation than I gave it.  Basically we no longer will drain the pipeline. When we `run_pipeline`, what comes out is a pipeline that's still lazy and ready to be processed.

This puts the requirement to process ctrl-c (and the stream itself) in other places in the code.

eg) 
https://github.com/nushell/nushell/pull/1359/files/f549c755cd92f84b5b04f7e675c355be292e4c9b#diff-dba02791b4baf8a328939029ac920005R654

**Pull request merged**